File: Samples/petstore/Python/swaggerpetstore/models/pet.py
# coding=utf-8
# --------------------------------------------------------------------------
# --------------------------------------------------------------------------

from msrest.serialization import Model


class Pet(Model):
    """A pet.

    A group of properties representing a pet.

    :param id: The id of the pet. A more detailed description of the id of
     the pet.
    :type id: long
    :param category:
    :type category: :class:`Category <petstore.models.Category>`
    :param name:
    :type name: str
    :param photo_urls:
    :type photo_urls: list of str
    :param tags:
    :type tags: list of :class:`Tag <petstore.models.Tag>`
    :param status: pet status in the store. Possible values include:
     'available', 'pending', 'sold'
    :type status: str
    """ 

    _validation = {
        'name': {'required': True},
        'photo_urls': {'required': True},
    }

    _attribute_map = {
        'id': {'key': 'id', 'type': 'long'},
        'category': {'key': 'category', 'type': 'Category'},
        'name': {'key': 'name', 'type': 'str'},
        'photo_urls': {'key': 'photoUrls', 'type': '[str]'},
        'tags': {'key': 'tags', 'type': '[Tag]'},
        'status': {'key': 'status', 'type': 'str'},
    }

    def __init__(self, name, photo_urls, id=None, category=None, tags=None, status=None):
        self.id = id
        self.category = category
        self.name = name
        self.photo_urls = photo_urls
        self.tags = tags
        self.status = status

File: Samples/petstore/NodeJS/models/pet.js
/*
 */

'use strict';

var models = require('./index');

var util = require('util');

/**
 * @class
 * Initializes a new instance of the Pet class.
 * @constructor
 * @summary A pet
 *
 * A group of properties representing a pet.
 *
 * @member {number} [id] The id of the pet. A more detailed description of the
 * id of the pet.
 *
 * @member {object} [category]
 *
 * @member {number} [category.id]
 *
 * @member {string} [category.name]
 *
 * @member {string} name
 *
 * @member {array} photoUrls
 *
 * @member {array} [tags]
 *
 * @member {string} [status] pet status in the store. Possible values include:
 * 'available', 'pending', 'sold'
 *
 */
function Pet() {
}

/**
 * Defines the metadata of Pet
 *
 * @returns {object} metadata of Pet
 *
 */
Pet.prototype.mapper = function () {
  return {
    required: false,
    serializedName: 'Pet',
    type: {
      name: 'Composite',
      className: 'Pet',
      modelProperties: {
        id: {
          required: false,
          serializedName: 'id',
          type: {
            name: 'Number'
          }
        },
        category: {
          required: false,
          serializedName: 'category',
          type: {
            name: 'Composite',
            className: 'Category'
          }
        },
        name: {
          required: true,
          serializedName: 'name',
          type: {
            name: 'String'
          }
        },
        photoUrls: {
          required: true,
          serializedName: 'photoUrls',
          type: {
            name: 'Sequence',
            element: {
                required: false,
                serializedName: 'StringElementType',
                type: {
                  name: 'String'
                }
            }
          }
        },
        tags: {
          required: false,
          serializedName: 'tags',
          type: {
            name: 'Sequence',
            element: {
                required: false,
                serializedName: 'TagElementType',
                type: {
                  name: 'Composite',
                  className: 'Tag'
                }
            }
          }
        },
        status: {
          required: false,
          serializedName: 'status',
          type: {
            name: 'String'
          }
        }
      }
    }
  };
};

module.exports = Pet;
